client/components/wcc-settings-form/index.js

```diff
@@ -9,19 +9,76 @@ import { successNotice, errorNotice } from 'state/notices/actions';
 import * as SettingsActions from 'state/settings/actions';
 import * as PackagesActions from 'state/form/packages/actions';
 import { getFormErrors } from 'state/selectors';
+import { translate as __ } from 'lib/mixins/i18n';
+import FormButton from 'components/forms/form-button';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
 
 const WCCSettingsForm = ( props ) => {
+	const currentStep = props.form.currentStep;
+	const currentStepLayout = props.layout[ currentStep ];
+
+	const renderCurrentStep = () => {
+		if ( ! currentStepLayout ) {
+			return <span>... spinner or something ...</span>;
+		}
+
+		return (
+			<WCCSettingsStep
+				{ ...props }
+				layout={ currentStepLayout }
+			/>
+		);
+	};
+
+	const renderActionButton = () => {
+		// TODO: extend <SaveForm> and use it in place of this
+		const label = ( currentStepLayout || {} ).action_label || __( 'Next' );
+		return (
+			<FormButtonsBar>
+				<FormButton type="button" onClick={ props.formActions.nextStep }>
+					{ label }
+				</FormButton>
+			</FormButtonsBar>
+		)
+	};
+
+	const renderTab = ( label, index ) => {
```

Comment:
Probably all of this `renderWhatever` methods should be its own component. I still don't know how fine-grained to go with stuff like this. But I haven't given this much thought, 95% of this code will need to be rewritten for sure :)